src/Core/Domain/Manufacturer/Command/Handler/AddManufacturerInterface.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Manufacturer\Command\Handler;
+
+
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Command\AddManufacturerCommand;
+
+interface AddManufacturerInterface
```

Comment:
Normally you would create a single interface for _all_ handlers and then add some sort of command bus where you can deposit any command and have it handled automatically. There you can also add concerns like command validation etc.

Reply:
If all handlers had the same interface, then it would be possible to use any handler to handle any command. I'm thinking about tactician for command bus: http://tactician.thephpleague.com/